locale/localization.js

```diff
@@ -0,0 +1,31 @@
+
+module.exports = {
+	addTranslationFunc : function (localeData){
+		// Add translation function to String prototype
+		String.prototype.translate = function(_groups = String.prototype.getTranslationDefaults()) {
+			const text = this.valueOf();
+			const groups = _groups ? Array.from(_groups) : [];
+			let obj = localeData;
+			while (groups?.length > 0) {
+				if(obj[groups[0]]) {
+					obj = obj[groups[0]];
+					groups.shift();
+					continue;
+				}
+				break;
+			}
+
+			if(obj[text] && obj[text].length > 0) return `${obj[text]}`;
+			return localeData[text]?.length > 0 ? `${localeData[text]}` : `${text}`;
+		};
+
+		// Add defaults
+		String.prototype.getTranslationDefaults = function() {
```

Comment:
I will just comment that I am not a huge fan of this `groups` parameter. It seems to add a layer of complexity that isn't really needed. Can I suggest we instead just handle the different translation groups via `group.key` notation?

`'nav.save'.translate()`